python/tests/sklearn/preprocessing/test_one_hot_encoder.py

```diff
@@ -0,0 +1,145 @@
+
+        ohe = OneHotEncoder(handle_unknown='error')
+        ohe.mlinit(prior_tf=self.le, output_features='{}_one_hot_encoded'.format(self.le.output_features))
+        ohe.fit(self.oh_data)
+
+        with self.assertRaises(ValueError):
+            ohe.serialize_to_bundle(self.tmp_dir, ohe.name)
+
+    def test_one_hot_encoder_serialization_fails_on_an_invalid_category_range(self):
+        self.oh_data[2][0] = 3  # make invalid category range
+
+        ohe = OneHotEncoder(handle_unknown='error')
+        ohe.mlinit(prior_tf=self.le, output_features='{}_one_hot_encoded'.format(self.le.output_features))
+        ohe.fit(self.oh_data)
+
+        with self.assertRaises(ValueError):
+            ohe.serialize_to_bundle(self.tmp_dir, ohe.name)
+
+    def test_one_hot_encoder_serialization_fails_when_using_the_drop_param(self):
+        ohe = OneHotEncoder(handle_unknown='error', drop='first')  # try to use `drop` parameter
+        ohe.mlinit(prior_tf=self.le, output_features='{}_one_hot_encoded'.format(self.le.output_features))
+        ohe.fit(self.oh_data)
+
+        with self.assertRaises(ValueError):
+            ohe.serialize_to_bundle(self.tmp_dir, ohe.name)
+
+    def test_one_hot_encoder_serialization_fails_when_using_the_dtype_param(self):
+        ohe = OneHotEncoder(handle_unknown='error', dtype=int)  # try to use `dtype` parameter
+        ohe.mlinit(prior_tf=self.le, output_features='{}_one_hot_encoded'.format(self.le.output_features))
+        ohe.fit(self.oh_data)
+
+        with self.assertRaises(ValueError):
+            ohe.serialize_to_bundle(self.tmp_dir, ohe.name)
+
+    def test_one_hot_encoder_serialization_succeeds_when_handle_unknown_is_set_to_error(self):
+        ohe = OneHotEncoder(handle_unknown='error')
+        ohe.mlinit(prior_tf=self.le, output_features='{}_one_hot_encoded'.format(self.le.output_features))
+        ohe.fit(self.oh_data)
+
+        ohe.serialize_to_bundle(self.tmp_dir, ohe.name)
+        with open("{}/{}.node/model.json".format(self.tmp_dir, ohe.name)) as json_data:
+            model = json.load(json_data)
+
+        self.assertEqual('one_hot_encoder', model['op'])
+        self.assertEqual(3, model['attributes']['size']['long'])
+        self.assertEqual('error', model['attributes']['handle_invalid']['string'])
+        self.assertEqual(False, model['attributes']['drop_last']['boolean'])
+
+    def test_one_hot_encoder_deserialization_succeeds_when_handle_unknown_is_set_to_error(self):
+        ohe = OneHotEncoder(handle_unknown='error')
+        ohe.mlinit(prior_tf=self.le, output_features='{}_one_hot_encoded'.format(self.le.output_features))
+        ohe.fit(self.oh_data)
+
+        ohe.serialize_to_bundle(self.tmp_dir, ohe.name)
+
+        node_name = "{}.node".format(ohe.name)
+        ohe_ds = OneHotEncoder()
+        ohe_ds.deserialize_from_bundle(self.tmp_dir, node_name)
+
+        self.oh_data[2][0] = 3  # Add an unknown category
```

Comment:
The problem here is that I need `oh_data` to form an invalid range. Like this:
`[0, 1, 2, 4]`
But if I were to pass strings through the `LabelEncoder`, it will always map them to a valid category range. Like this:
`[0, 1, 2, 3]`
So this test is meant for the case where somebody loads a csv and tries to one-hot encode an integer feature. Say number of bedrooms in a house. No house has zero bedrooms, so we would force them to convert the data to category indices before one-hot encoding